src/convenience/webhook.ts

```diff
@@ -34,8 +34,89 @@ export interface WebhookOptions {
+type WebhookAdapter<
+    C extends Context = Context,
+    A extends Adapters[AdapterNames] = Adapters[AdapterNames],
+> = {
+    (
+        bot: Bot<C>,
+        webhookOptions?: WebhookOptions,
+    ): Adapter<A>;
+    (
+        bot: Bot<C>,
+        onTimeout?: WebhookOptions["onTimeout"],
+        timeoutMilliseconds?: WebhookOptions["timeoutMilliseconds"],
+        secretToken?: WebhookOptions["secretToken"],
+    ): Adapter<A>;
+};
+
+function createWebhookAdapter<
+    C extends Context = Context,
+    A extends Adapters[AdapterNames] = Adapters[AdapterNames],
+>(adapter: A): WebhookAdapter<C, A> {
+    return (
+        bot: Bot<C>,
+        onTimeout?:
+            | WebhookOptions
+            | WebhookOptions["onTimeout"],
+        timeoutMilliseconds?: WebhookOptions["timeoutMilliseconds"],
+        secretToken?: WebhookOptions["secretToken"],
+    ) => {
+        const {
+            onTimeout: timeout = "throw",
+            timeoutMilliseconds: ms = 10_000,
+            secretToken: token,
+        } = typeof onTimeout === "object"
+            ? onTimeout
+            : { onTimeout, timeoutMilliseconds, secretToken };
+
+        return webhookCallback(bot, adapter, timeout, ms, token);
+    };
+}
+// TODO: add docs examples for each adapter?
+/**
+ * Contains factories of callback function that you can pass to a web framework
+ * (such as express) if you want to run your bot via webhooks. Use it like this:
+ * ```ts
+ * const app = express() // or whatever you're using
+ * const bot = new Bot('<token>')
+ *
+ * app.use(webhookAdapters.express(bot))
+ * ```
+ *
+ * Confer the grammY
+ * [documentation](https://grammy.dev/guide/deployment-types) to read more
+ * about how to run your bot with webhooks.
+ *
+ * @param bot The bot for which to create a callback
+ * @param webhookOptions Further options for the webhook setup
+ */
+export const webhookAdapters = {
+    awsLambda: createWebhookAdapter(adapters.awsLambda),
+    awsLambdaAsync: createWebhookAdapter(adapters.awsLambdaAsync),
+    azure: createWebhookAdapter(adapters.azure),
+    bun: createWebhookAdapter(adapters.bun),
+    cloudflare: createWebhookAdapter(adapters.cloudflare),
+    cloudflareModule: createWebhookAdapter(adapters.cloudflareModule),
+    express: createWebhookAdapter(adapters.express),
+    fastify: createWebhookAdapter(adapters.fastify),
+    hono: createWebhookAdapter(adapters.hono),
+    http: createWebhookAdapter(adapters.http),
+    https: createWebhookAdapter(adapters.http),
+    koa: createWebhookAdapter(adapters.koa),
+    nextJs: createWebhookAdapter(adapters.nextJs),
+    nhttp: createWebhookAdapter(adapters.nhttp),
+    oak: createWebhookAdapter(adapters.oak),
+    serveHttp: createWebhookAdapter(adapters.serveHttp),
+    stdHttp: createWebhookAdapter(adapters.stdHttp),
+    sveltekit: createWebhookAdapter(adapters.sveltekit),
+    worktop: createWebhookAdapter(adapters.worktop),
+    callback: createWebhookAdapter(adapters.callback),
```

Comment:
I'd use getters to avoid unnecessary `createWebhookAdapter` calls at startup.

Reply:
nice catch!